### PR TITLE
Add cover art support to Discogs, FanArtTV, and MusicBrainz

### PR DESCRIPTION
**Pull request by aw-was-here**
* Discogs: fetch front cover art from release listings
* FanArtTV: fetch album cover art when available
* MusicBrainz: cover art already supported; keep coverimageraw when title-strip fallback match is used (album may differ but art is still valid)
* processors.py: run tinytag/hostmeta before imagecache lookup so embedded file art always wins over cached URL covers; bytes sources always beat cached URL covers
* static_handlers.py: fix _fetch_cover_url to use resp.read() instead of resp.content.read(n) which returned only the current aiohttp buffer (~16 KB); affects EarShot and 2PC remote setups that send HTTP cover URLs
* discordbot.py: guard change_presence with is_ready() to avoid AttributeError when Discord WebSocket is not connected
* Docs: update features.md, llms.txt, discogs.md, fanarttv.md, and musicbrainz.md to reflect cover art additions

## Summary by Sourcery

Add configurable album cover art support for Discogs, FanArtTV, and MusicBrainz artist extras and recognition, including UI controls, docs, and tests.

New Features:
- Queue Discogs album cover images from release search results as front cover art when enabled.
- Queue FanArtTV album cover images from the albums section of API responses as front cover art when enabled.
- Allow MusicBrainz recognition to fetch album cover art from Cover Art Archive controlled by a new cover art setting.

Enhancements:
- Extend Discogs, FanArtTV, and MusicBrainz settings UIs with cover art checkboxes and defaults to control cover fetching behavior.
- Document album cover art capabilities for Discogs, FanArtTV, and MusicBrainz in feature and provider documentation.

Tests:
- Add parameterized tests to verify cover art behavior is gated by configuration for Discogs, FanArtTV, and MusicBrainz, including edge cases without album IDs.

## Summary by Sourcery

Add configurable album cover art fetching across Discogs, FanArtTV, and MusicBrainz, ensuring embedded/tag-based images take precedence and improving remote cover URL handling and Discord bot stability.

New Features:
- Queue Discogs release cover images as front cover art when enabled in settings.
- Queue FanArtTV album cover images as front cover art using MusicBrainz release group IDs when enabled.
- Allow MusicBrainz recognition to fetch album cover art from Cover Art Archive controlled by a new cover art setting.

Enhancements:
- Run hostmeta/tinytag/image2png processors before image cache lookups so embedded and byte-based covers override cached URL images.
- Expose new cover art toggles in Discogs, FanArtTV, and MusicBrainz settings UIs and track MusicBrainz release group IDs throughout metadata and storage.
- Improve remote cover URL fetching to read full responses within size limits and add logging for cover art retrieval outcomes.
- Keep MusicBrainz-derived cover art in fallback/title-strip scenarios instead of clearing it.
- Guard Discord presence updates on bot readiness to avoid errors when the WebSocket is disconnected.

Documentation:
- Document cover art capabilities and configuration for Discogs, FanArtTV, and MusicBrainz in feature and provider documentation, including template variables for MusicBrainz release group IDs.

Tests:
- Add Discogs, FanArtTV, and MusicBrainz tests to validate cover art configuration gating, album-cover queuing behavior, and database schema expectations around new metadata fields.

### CODE REVIEW

File: nowplaying/webserver/static_handlers.py [hunk 1]
```diff
@@ -473,7 +473,11 @@ async def _fetch_cover_url(self, request: web.Request, url: str) -> bytes | None
                         "cover URL Content-Length %d exceeds limit: %s", content_length, url
                     )
                     return None
-                return await resp.content.read(self.MAX_COVER_BYTES)
+                data = await resp.read()
+                if len(data) > self.MAX_COVER_BYTES:
+                    logging.warning("cover URL body %d bytes exceeds limit: %s", len(data), url)
+                    return None
+                return data
```
[the remark this answers is not in the document]
Comment (aw-was-here) [reply nowplaying/webserver/static_handlers.py:480]: Except that's not how aiohttp read() works.  read() appears to dump just the buffer contents, not waiting for the full read to finish.
